pkg/codegen/templates/client.tmpl

```diff
@@ -18,9 +18,9 @@ type Client struct {
 	// customized settings, such as certificate chains.
 	Client HttpRequestDoer
 
-	// A callback for modifying requests which are generated before sending over
+	// A list of callbacks for modifying requests which are generated before sending over
 	// the network.
-	RequestEditor RequestEditorFn
+	RequestEditors []RequestEditorFn
```

Comment:
This is a breaking change. Anyone accessing the previous RequestEditor will get a compile error now.

We don't need a list of RequestEditors on the client, could you leave it as a single RequestEditor?

Your per-handler request editors supercede this global per-client editor.

Reply:
> We don't need a list of RequestEditors on the client, could you leave it as a single RequestEditor?

But how would then one register multiple RequestEditors with a client?

I mean, modifying `RequestEditors` directly should be a bad pattern anyway, no? You have to modify it through constructor options.

But if this is really the problem, we could have `RequestEditor` and `RequestEditors`, where first the `RequestEditor` is run an then new `RequestEditors` in order.

Reply:
I'm guessing the breakage from this will be minimal, I will merge as-is.

The reason that I originally had a single request editor is that you yourself can call a bunch from within it, so it seemed redundant.

Your code never uses that editor array on the client, as you use the list specified on the handler, and only of present, do you append. So, the change from 1 to many doesn't really gain much.